src/components/PuzzleList/PuzzleList.js

```diff
@@ -134,7 +134,17 @@ export default class PuzzleList extends PureComponent {
 
   render() {
     if (localStorage.newPuzzleListEnabled) {
-      return <NewPuzzleList />;
+      const filter = {
+        nameOrTitleFilter: this.props.search,
+        sizeFilter: this.props.sizeFilter,
+      };
+      return (
+        <NewPuzzleList
+          filter={filter}
```

Comment:
unrelated, but should force a remount of the component when the user creates a new public puzzle